packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol

```diff
@@ -33,6 +68,36 @@ contract FuelMessagePortalV3 is FuelMessagePortalV2 {
         messageIsBlacklisted[messageId] = false;
     }
 
+    /**
+     * @notice Resets the rate limit amount.
+     * @param _amount The amount to reset the limit to.
+     */
+    function resetRateLimitAmount(uint256 _amount) external onlyRole(SET_RATE_LIMITER_ROLE) {
+        uint256 withdrawalLimitAmountToSet;
+        bool amountWithdrawnLoweredToLimit;
+        bool withdrawalAmountResetToZero;
+        
+        // if period has expired then currentPeriodAmount is zero
+        if (currentPeriodEnd < block.timestamp) {
+            unchecked {
+                currentPeriodEnd = block.timestamp + rateLimitDuration;
+            }
+            withdrawalAmountResetToZero = true;
+        } else {
+            // If the withdrawn amount is higher, it is set to the new limit amount
+            if (_amount < currentPeriodAmount) {
+                withdrawalLimitAmountToSet = _amount;
+                amountWithdrawnLoweredToLimit = true;
+            }
+        }
+
+        limitAmount = _amount;
+
+        if (withdrawalAmountResetToZero || amountWithdrawnLoweredToLimit) {
+            currentPeriodAmount = withdrawalLimitAmountToSet;
+        }
+    }
+
```

Comment:
yeah that is a known risk and that's why we do have https://github.com/FuelLabs/fuel-canary-watchtower in place to handle & monitor such situations and this would be a known risk with any generic rate limit impl too